### PR TITLE
[2.19.x] DDF-5716 G-7628 Add support for specifying UTM coords with the hemisphere notation in the gazetteer

### PR DESCRIPTION
**Pull request by mojogitoverhere**
# Backport of https://github.com/codice/ddf/pull/5718

#### What does this PR do?
When a user enters a UTM coordinate in the gazetteer search box with no latitude band, `N` is used as the default. This behavior makes is appear as if we support specifying the Northern or Southern hemisphere in place of the latitude band since the gazetteer correctly pans to the Northern hemisphere point. However, this is only a side-effect of how we are parsing the coordinates. In reality, `N` does not directly correspond to the Northern hemisphere. Instead, if the latitude band does not actually correspond with the northing value, then the invalid latitude band is used to *indirectly* determine which hemisphere to use with C-M latitude bands corresponding to the Southern hemisphere and N-X latitude bands corresponding the the Northern hemisphere. As a result, the default behavior of adding the `N` latitude band happens to correctly takes the user to the Northern hemisphere coordinate. But when they try to use the `S` latitude band as the Southern hemisphere equivalent, they are taken to the same coordinate as the `N` latitude band because `S` also falls between N-X.

This PR allows users to use the `S` latitude band by showing two suggestions for that case:
1. the usual coordinate with the `S` latitude band, and
2. the Southern hemisphere alternative

For consistency and to teach the users about this feature, two suggestions are also shown for coordinates with the `N` latitude band even though both actually point to the same location.

In addition, if no latitude band is entered then both the Northern hemisphere and Southern hemisphere suggestions are shown.

#### Who is reviewing it? 

@brianfelix @brhumphe @kentmorrissey @mazarag2 @Lambeaux 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@millerw8

#### How should this be tested?

1. Start solr by running `docker-compose up` in distribution/docker/solrcloud
2. Start DDF and install the standard profile (`profile:install standard`)
3. In the admin UI, go to "Catalog UI Search" and unselect "Use Online Gazetteer"
4. In the gazetteer search box in Intrigue, test these cases. Make sure the suggestions are shown as expected and the map pans to the correct location. You can use this tool (https://mappingsupport.com/p2/coordinates-utm-gissurfer-maps.html) to determine if the locations are correct and get new coordinates to test (right clicking on that map shows the UTM coordinates for that point).

| Gazetteer Query        | UTM suggestion(s)           | Location |
| ------------- |-------------|-------------|
| 17R 461652mE 2993043mN      | UTM: [ 17R 461652mE 2993043mN ] | Florida
| 19G 577867mE 5117939mN      | UTM: [ 19G 577867mE 5117939mN ]      | Argentina
| 34 678879 6697434 | UTM: [ 34  678879mE 6697434mN N ], UTM: [ 34  678879mE 6697434mN N ]      | Finland, South Africa
| 53S 571222mE 7287229mN | UTM: [ 53 570673mE 7181387mN N ], UTM: [ 53 570673mE 7181387mN N ]      | Russia, Australia
| 20N 235789mE 756626mN | UTM: [ 20N 235789mE 756626mN ], UTM: [ 20 235789mE 756626mN N ] | Both go to Venezuela      | 
| 10U 432103mE 5412830mN 19R 430147mE 2862469mN | UTM: [ 10U 432103mE 5412830mN 19R ] [ 430147mE 2862469mN ]      | The extent of the US

5. Enter combinations of UTM and UPS coordinates
     - Verify if only UTM coordinates are entered, the suggestion prefix is `UTM`
     - Verify if only UPS coordinates are entered, the suggestion prefix is `UPS`
     - Verify if a mix of UTM and UPS coordinates are entered, the suggestion prefix is `UTM/UPS`
5. Try any other coordinates / combinations of coordinates
6. Try some malformed coordinates (i.e. no zone, no latitude band, no units, multiple spaces, etc)

#### What are the relevant tickets?
Fixes: #5716 

#### Screenshots
S latitude band and Southern hemisphere suggestions
single-UTM-point

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by mojogitoverhere** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java [hunk 1]
```diff
@@ -76,42 +79,178 @@ public UtmUpsCoordinateProcessor(CoordinateSystemTranslator translator) {
...
+      nameBuilder.append("UPS: [ ").append(utmUps.toString()).append(" ]");
+    }
+    return nameBuilder.toString();
+  }
+
+  private static String makeSuggestionText(List<UtmUpsCoordinate> utmUpsList) {
+    final StringBuilder nameBuilder = new StringBuilder();
+    long numberOfUtmCoords = utmUpsList.stream().filter(UtmUpsCoordinate::isUTM).count();
+    int total = utmUpsList.size();
+    if (numberOfUtmCoords == total) {
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java]: ❗️ This is not going to accomplish exactly what you want. The `isUTM` and `isUPS` functions tell you if the coordinate "data" exists in the "space", not what the text format is. On top of that, there is overlap, so both functions could return true. They are not mutually exclusive. 

See my comments on my review of the PR against master: https://github.com/codice/ddf/pull/5718#pullrequestreview-342925894

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java [hunk 2]
```diff
@@ -76,42 +79,178 @@ public UtmUpsCoordinateProcessor(CoordinateSystemTranslator translator) {
...
+    List<LatLon> latLonList =
+        utmUpsList.stream().map(this::latLonFromUtmUtps).filter(Objects::nonNull).collect(toList());
+    String suggestionText = makeSuggestionText(utmUpsList);
+
+    return new LiteralSuggestion(LITERAL_SUGGESTION_ID, suggestionText, latLonList);
+  }
+
+  private static String makeSuggestionText(UtmUpsCoordinate utmUps) {
+    final StringBuilder nameBuilder = new StringBuilder();
+    if (utmUps.isUTM()) {
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java]: ❗️ Same here. This isn't exactly what you want.

**Comment by mojogitoverhere** [comment]
@Lambeaux Since the master counterpart was already merged, let's merge this one as-is and I'll address your comments in a follow up PR to keep things consistent across the branches

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java (hunk 1 above)
Comment (Lambeaux) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java]: ✅ After further analysis we could not reproduce the problem. Turns out the semantics of `UtmUpsCoordinate` are different from the other coordinate impls. This is fine.

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java (hunk 2 above)
Comment (Lambeaux) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java]: ✅ After further analysis we could not reproduce the problem. Turns out the semantics of `UtmUpsCoordinate` are different from the other coordinate impls. This is fine.

**Pull request merged**